articles/qos_deadline_liveliness_lifespan.md

```diff
@@ -0,0 +1,188 @@
+
+For example, it doesn't do much good to specify deadline policy via the vendor configuration mechanism if your ROS application is never notified when a publisher violates the deadline.
+In such cases the application needs to be able to respond to these events in order to log failures, notify users, or adjust system behavior.
+
+When interaction with the application is needed it is also preferable to have a ROS defined interface as opposed to having the application program directly against the DDS vendor API.
+Many nodes in ROS are written to be shared and reused by the community.
+Since different end users will want to use different middleware vendors for their applications, it is important to maintain that flexibility for nodes that want to take advantage of these additional QoS policies.
+
+**Based on this criteria, native ROS support should be provided for Deadline and Liveliness but not for the Lifespan policy.**
+While Deadline and Liveliness will both have events that the application will initiate and need to be informed of, the Lifespan policy can be entirely handled by the underlying DDS service without the need for any intervention by the application.
+
+### ROS Changes
+
+These are the various changes that would be needed within ROS in order to natively support Deadline and Liveliness.
+
+#### Topic Status Event Handler
+
+Both the Deadline and Liveliness policies generate events from the DDS service layer that the application will need to be informed of.
+For Deadlines, the subscriber receives event notifications if it doesn't receive anything within the deadline and the publisher receives event notifications if it doesn't publish anything within the deadline.
+For Liveliness, subscribers receive events when the publisher they're listening to is no longer alive. Both of these fall under a category of "Topic Status Events". 
+
+To handle these notifications, a new callback function can be provided by the user that will be called any time a `TopicStatusEvent` occurs for a particular topic.
+It will receive as a parameter a `TopicStatusEvent` enum value that specifies what type of event took place and a timestamp of when the event took place.
+This callback function would be provided by the user's application when it calls the create function for publishers and subscribers.
+The constructors and create functions will be overloaded to make this new handler optional.
+
+The choice to use a callback function as opposed to another notification mechanism is to remain consistent with the existing ROS interface and not add a new method of waiting on ROS data/events.
+
+#### QoS Struct
+
+Minimal changes will need to be made to the QoS struct that is passed into the creation functions for Topics, Services, and Actions.
+A couple new enum values will be added for the Deadline and Liveliness settings and then a couple integers will be added to specify the time values for these policies.
```

Comment:
"couple of new enum values"
"couple of integers"

Although more technically exact language would be nice.

Reply:
I've added more detailed descriptions of the enum settings and how they would apply to Topics and Services. The names aren't necessarily finalized, but that should provide more detail.